tests/scala/org/pantsbuild/zinc/compiler/JarCreationSpec.scala

```diff
@@ -0,0 +1,41 @@
+package org.pantsbuild.zinc.compiler
+
+import java.io.File
+import java.nio.file.{Path, Paths}
+import scala.collection.mutable
+
+import sbt.io.IO
+
+import org.junit.runner.RunWith
+import org.scalatest.WordSpec
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.MustMatchers
+
+@RunWith(classOf[JUnitRunner])
+class JarCreationSpec extends WordSpec with MustMatchers {
+  "JarCreationWithoutClasses" should {
+    "succeed when input classes are provided" in {
+      IO.withTemporaryDirectory { tempDir =>
+        val filePaths = new mutable.TreeSet[Path]()
+        val jarOutputPath = Paths.get(tempDir.toString, "spec-empty-output.jar")
+        OutputUtils.createJar(filePaths, jarOutputPath, System.currentTimeMillis())
+        OutputUtils.existsClass(jarOutputPath, "NonExistent.class") must be(false)
+      }
+    }
+  }
+  "JarCreationWithClasses" should {
+    "succeed when input classes are provided" in {
+      IO.withTemporaryDirectory { tempDir =>
+
+        val tempDirPath = tempDir.toString
+        val tempFile = File.createTempFile(tempDirPath, "Clazz.class")
+        val filePaths = mutable.TreeSet(tempFile.toPath)
+        val jarOutputPath = Paths.get(tempDirPath.toString, "spec-valid-output.jar")
+
+        OutputUtils.createJar(filePaths, jarOutputPath, System.currentTimeMillis())
+        OutputUtils.existsClass(jarOutputPath, tempFile.toString) must be(true)
```

Comment:
Thanks for writing the test. Things are a bit clearer to me now.

I think createJar should take a root for the input file paths, that way the entry paths can be relative to that root.

for example, if you had an input dir like `temp-dir` that looked like this.
```
temp-dir/org/example/Clazz.class
temp-dir/org/example/inner/Clazz2.class
```

I think you'd want the resulting jar to contain

```
org/example/Clazz.class
org/example/inner/Clazz2.class
```

I'm not sure it does this right now.

Does that make sense? I might be missing something.